controllers/argocd/role.go

```diff
@@ -156,6 +181,91 @@ func (r *ReconcileArgoCD) reconcileRole(name string, policyRules []v1.PolicyRule
 	return roles, nil
 }
 
+func (r *ReconcileArgoCD) reconcileRoleForSupportedNamespaces(name string, policyRules []v1.PolicyRule, cr *argoprojv1a1.ArgoCD) ([]*v1.Role, error) {
+	var roles []*v1.Role
+
+	// create policy rules for each supported namespace for ArgoCD Server
+	for _, sourceNamespace := range cr.Spec.SourceNamespaces {
+
+		namespace := &corev1.Namespace{}
+		if err := r.Client.Get(context.TODO(), types.NamespacedName{Name: sourceNamespace}, namespace); err != nil {
+			return nil, err
+		}
+
+		managedNamespace := false
+		// do not reconcile roles for namespaces already containing managed-by label
+		// as it already contains roles reconciled during reconcilation of ManagedNamespaces
+		for _, ns := range r.ManagedNamespaces.Items {
+			if reflect.DeepEqual(ns, namespace) {
+				managedNamespace = true
+				break
+			}
+		}
+		if managedNamespace {
+			continue
+		}
+
+		log.Info(fmt.Sprintf("Reconciling role for %s", namespace.Name))
+
+		role := newRoleForSupportedNamespaces(name, namespace.Name, policyRules, cr)
+		if err := applyReconcilerHook(cr, role, ""); err != nil {
+			return nil, err
+		}
+		role.Namespace = namespace.Name
+		existingRole := v1.Role{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: role.Name, Namespace: namespace.Name}, &existingRole)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return nil, fmt.Errorf("failed to reconcile the role for the service account associated with %s : %s", name, err)
+			}
+			log.Info(fmt.Sprintf("Fetching Role Error: %s", err.Error()))
+		}
+
+		// do not reconcile roles for namespaces already containing managed-by-cluster-argocd label
+		// as it already contains roles reconciled during reconcilation of ManagedNamespaces
+		if _, ok := r.ManagedSourceNamespaces[sourceNamespace]; ok {
+			// If sourceNamespace includes the name but role is missing in the namespace, create the role
+			if reflect.DeepEqual(existingRole, v1.Role{}) {
```

Comment:
This should  be 
```
if !reflect.DeepEqual(existingRole, v1.Role{}) { ... }
```

Reply:
If we were not able to find the existingRole in line 216, it errors out and continues the operations to create the necessary role by checking that the value of existingRole matches the value `v1.Role{}` (value set at line 215). Thus I believe this part of code does not need to change.

Reply:
oops yeah you're right 
sorry ignore that